pkg/agent/nplagent/rules/iptable_rule.go

```diff
@@ -0,0 +1,190 @@
+// +build !windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/coreos/go-iptables/iptables"
+	"k8s.io/klog"
+)
+
+const NodePortLocalChain = "NODE-PORT-LOCAL"
+
+type IPTableRules struct {
+	name  string
+	table *iptables.IPTables
+}
+
+func NewIPTableRules() *IPTableRules {
+	iptables.New()
+	ipt, err := iptables.New()
+	if err != nil {
+		klog.Infof("init iptable for NPL failed: %v\n", err)
+		return nil
+	}
+	iptRule := IPTableRules{
+		name:  "NPL",
+		table: ipt,
+	}
+	return &iptRule
+}
+
+var once sync.Once
+
+func (ipt *IPTableRules) Init() bool {
+	if !ipt.DeleteAllRules() {
+		return false
+	}
+	return ipt.CreateChains()
+}
+
+// CreateChains : Create the chain NODE-PORT-LOCAL in NAT table
+// All DNAT rules for NPL would be added in this chain
+func (ipt *IPTableRules) CreateChains() bool {
```

Comment:
Could you check if makes sense to leverage any funcs defined in pkg/iptables? Like EnsureRule, EnsureChain?

Reply:
Sure I'll check it out